content/docs/user-guide/testing/parallel-pattern/_index.md

```diff
@@ -0,0 +1,476 @@
+* `--editor-parallel`: Overrides the number of maximum editors run in parallel. Setting this option to `1` is equivalent to `--no-editor-parallel`.
+
+## Best practices
+
+* **Don’t create a level unless necessary**. Use an already existing empty level and don’t save the changes once the test is over.
+* **Tests should be be self-contained**. No external tools should determine if the test has passed or failed. Instead, the Editor should do this work. Use the one of the Editor return codes as result of the test, `0x0` for success and `0xF` for failure.
+* **Don’t read the Editor log file directly**. To read Editor log information else use the `DebugTraceBus` and `Tracer` utilities. The `DebugTraceBus` is able to capture any of these, without having to create an I/O dependency on a file.
+    
+    ```python
+with Tracer() as section_tracer:
+    # Do stuff
+
+# section_tracer now contains all prints,errors and warnings that happened inside the block
+if section_tracer.has_errors:
+        assert False, "AZ_Errors happened during execution"
+```
+* **Tests must be deterministic **. The application will run at different FPS and take different amounts of time time loading assets. When running the test make sure it is deterministic with these conditions.
+* **Wait on conditions instead of waiting for time.** Some tests require performing an action and waiting for completion. To optimize the amount of time your test runs, and avoiding false negatives caused by timeouts, wait on a condition instead of for an arbitrary time:
+    ```python
+### BAD ###
+enable_gravity()
+# Wait 5 seconds for the entity to fall
+idle_wait(5.0)
+if entity.position.y >= 0:
+        assert False, "Entity didn't fall"
+
+
+### GOOD ###
+enable_gravity()
+def has_fallen():
+        return entity.position.y < 0
+# Wait for the entity to fall, a maximum time of 5 seconds
+# has_fallen() will be called in every frame as long it returns False
+helper.wait_for_condition(has_fallen, 5.0)
+if entity.position.y >= 0:
+        assert False, "Entity didn't fall"
+    ```
+* **Compare values within ranges.** Using ranges for values rather than equality, especially for float values, makes tests more likely to succeed. Vector and other math constructs have utility functions for detecting values within a specific range.
+    ```python
+### BAD ###
+if entity.position != azlmbr.math.Vector3(50.0, 50.0, 50.0):
+        assert False, "Entity is not in correct position"
+
+### GOOD ###
+if not entity.position.IsClose(azlmbr.math.Vector3(50.0, 50.0, 50.0)):
+        assert False, "Entity is not in correct position"
+    ```
+
+## Debugging tests
+
+During test development, it can be time-consuming to launch the Editor on a test-by-test basis. Instead, you can launch the Editor normally and then use the `pyRunFile` command from the Editor console:
```

Comment:
```suggestion
During test development, it can be time-consuming to launch O3DE Editor on a test-by-test basis. Instead, you can launch O3DE Editor normally and then use the `pyRunFile` command from **Console**:
```